freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/user/handler/SetPasswordHandler.java

```diff
@@ -76,4 +92,34 @@ public void accept(Event<SetPasswordRequest> setPasswordRequestEvent) {
             request.getResult().onError(e);
         }
     }
+
+    private void setPasswordHash(Stack stack, SetPasswordRequest request, FreeIpaClient freeIpaClient, WorkloadCredential workloadCredential)
+            throws IOException, FreeIpaClientException {
+        String accountId = Crn.fromString(stack.getEnvironmentCrn()).getAccountId();
+        boolean credentialsUpdateOptimizationEnabled = entitlementService.usersyncCredentialsUpdateOptimizationEnabled(accountId);
+        LOGGER.info("Credentials update optimization is{} enabled for account {}", credentialsUpdateOptimizationEnabled ? "" : " not", accountId);
+
+        if (credentialUpdateRequired(credentialsUpdateOptimizationEnabled, request.getUsername(), freeIpaClient, workloadCredential)) {
+            workloadCredentialService.setWorkloadCredential(credentialsUpdateOptimizationEnabled, freeIpaClient,
+                    new WorkloadCredentialUpdate(request.getUsername(), request.getUserCrn(), workloadCredential));
+        } else {
+            LOGGER.debug("Not setting workload credentials for user '{}' because credentials are already up to date", request.getUsername());
+        }
+    }
+
+    private boolean credentialUpdateRequired(boolean credentialsUpdateOptimizationEnabled, String username, FreeIpaClient freeIpaClient,
+            WorkloadCredential workloadCredential) throws FreeIpaClientException {
+        boolean updateCredentials = true;
```

Comment:
could you drop the variable and just use `return true` or `return false`

Reply:
Done.